ReactiveUI/Cocoa/CommonReactiveSource.cs

```diff
@@ -278,30 +234,74 @@ void resetupAll(IReadOnlyList<TSectionInfo> newSectionInfo)
             //
             // TODO: Instead of listening to Changed events and then reseting,
             // we could listen to more specific events and avoid some reloads.
-            disp.Add(sectionChanging.Subscribe(_ => {
+            disp.Add(sectionChanging.ObserveOn(ImmediateScheduler.Instance).Subscribe(_ => {
```

Comment:
Sorry for delay. Have made changes but I want to test before committing. Alas, my license recently expired. Even though I have renewed I need to refresh in XCode and it's unable to contact Apple servers atm.

![image](https://cloud.githubusercontent.com/assets/1901832/6768238/031b9738-d0b0-11e4-99ef-7b704fe622aa.png)

*sigh*